Apps/WebClient/src/NewClientApp/src/stores/note.ts

```diff
@@ -0,0 +1,197 @@
+import { ErrorSourceType, ErrorType } from "@/constants/errorType";
+import { ResultType } from "@/constants/resulttype";
+import { container } from "@/ioc/container";
+import { SERVICE_IDENTIFIER } from "@/ioc/identifier";
+import { ResultError } from "@/models/errors";
+import { LoadStatus, Operation, OperationType } from "@/models/storeOperations";
+import UserNote from "@/models/userNote";
+import { ILogger, IUserNoteService } from "@/services/interfaces";
+import { useErrorStore } from "@/stores/error";
+import { defineStore } from "pinia";
+import { computed, ref } from "vue";
+
+export const useNoteStore = defineStore("healthVisits", () => {
+    const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+    const noteService = container.get<IUserNoteService>(
+        SERVICE_IDENTIFIER.NoteService
+    );
+
+    const errorStore = useErrorStore();
+
+    // Refs
+    const notes = ref<UserNote[]>([]);
+    const status = ref(LoadStatus.NONE);
+    const statusMessage = ref<string>("");
+    const error = ref<ResultError>();
+    const lastOperation = ref<Operation | null>(null);
+
+    // Computed
+    const notesCount = computed(() => notes.value.length);
+    const noteAreLoading = computed(
+        () => status.value === LoadStatus.REQUESTED
+    );
+
+    // Mutations
+    function setNotesRequested() {
+        status.value = LoadStatus.REQUESTED;
+    }
+
+    function setRetrievedNotes(userNotes: UserNote[]) {
+        notes.value = userNotes;
+        error.value = undefined;
+        status.value = LoadStatus.LOADED;
+    }
+
+    function setCreatedNote(note: UserNote) {
+        lastOperation.value = new Operation(
+            note.id as string,
+            OperationType.ADD
+        );
+        notes.value.push(note);
+    }
+
+    function setUpdatedNote(note: UserNote) {
+        lastOperation.value = new Operation(
+            note.id as string,
+            OperationType.UPDATE
+        );
+    }
+
+    function setDeletedNote(note: UserNote) {
+        const noteIndex = notes.value.findIndex((x) => x.id === note.id);
+        if (noteIndex > -1) {
+            lastOperation.value = new Operation(
+                note.id as string,
+                OperationType.DELETE
+            );
+            notes.value.splice(noteIndex, 1);
+        }
+    }
+
+    function setNotesError(resultError: ResultError) {
+        error.value = resultError;
+        statusMessage.value = resultError.resultMessage;
+        status.value = LoadStatus.ERROR;
+    }
+
+    // Helpers
+    function handleError(
+        hdid: string,
+        resultError: ResultError,
+        errorType: ErrorType
+    ): void {
+        const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+
+        logger.error(`ERROR: ${JSON.stringify(resultError)}`);
```

Comment:
```suggestion
        logger.error(`ERROR: ${JSON.stringify(resultError)}`);
```

The logger has already been retrieved.